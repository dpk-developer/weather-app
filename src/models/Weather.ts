export interface WeatherResponse {
  name: string;
  main: { temp: number };
  weather: [{ description: string; icon: string }];
}
