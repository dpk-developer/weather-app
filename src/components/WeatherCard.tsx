import React from 'react';
import { View, Text, Image, StyleSheet } from 'react-native';

import { useTheme } from '../hooks/useTheme';
import { WeatherResponse } from '../models/Weather.ts';

interface Props {
  weather: WeatherResponse;
}

const WeatherCard = ({ weather }: Props) => {
  const { theme } = useTheme();

  const icon = `https://openweathermap.org/img/wn/${weather.weather[0].icon}@2x.png`;

  return (
    <View style={[styles.card, { backgroundColor: theme.card }]}>
      <Text style={[styles.city, { color: theme.text }]}>{weather.name}</Text>

      <Image source={{ uri: icon }} style={styles.icon} />

      <Text style={[styles.temp, { color: theme.text }]}>
        {Math.round(weather.main.temp)}Â°C
      </Text>

      <Text style={[styles.desc, { color: theme.text }]}>
        {weather.weather[0].description}
      </Text>
    </View>
  );
};

const styles = StyleSheet.create({
  card: {
    alignItems: 'center',
    padding: 16,
    borderRadius: 8,
    margin: 16,
    elevation: 2,
  },
  city: {
    fontSize: 24,
    fontWeight: 'bold',
  },
  icon: {
    width: 100,
    height: 100,
  },
  temp: {
    fontSize: 32,
    fontWeight: '500',
  },
  desc: {
    fontSize: 18,
    fontWeight: '500',
    textTransform: 'capitalize',
  },
});

export default WeatherCard;
