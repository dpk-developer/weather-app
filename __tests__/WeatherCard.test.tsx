import React from 'react';
import { render } from '@testing-library/react-native';

import WeatherCard from '../src/components/WeatherCard';
import { WeatherResponse } from '../src/models/Weather';

jest.mock('../src/hooks/useTheme', () => ({
  useTheme: () => ({
    theme: {
      text: '#000',
      card: '#fff',
    },
  }),
}));

const mockWeather: WeatherResponse = {
  name: 'London',
  main: { temp: 21 },
  weather: [{ description: 'clear sky', icon: '01d' }],
};

describe('WeatherCard', () => {
  it('renders weather info correctly', () => {
    const { getByText, toJSON } = render(<WeatherCard weather={mockWeather} />);
    expect(getByText('London')).toBeTruthy();
    expect(getByText('21Â°C')).toBeTruthy();
    expect(getByText('clear sky')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });
});
