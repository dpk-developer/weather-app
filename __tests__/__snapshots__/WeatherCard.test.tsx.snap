// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCard renders weather info correctly 1`] = `
<View
  style={
    [
      {
        "alignItems": "center",
        "borderRadius": 8,
        "elevation": 2,
        "margin": 16,
        "padding": 16,
      },
      {
        "backgroundColor": "#fff",
      },
    ]
  }
>
  <Text
    style={
      [
        {
          "fontSize": 24,
          "fontWeight": "bold",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    London
  </Text>
  <Image
    source={
      {
        "uri": "https://openweathermap.org/img/wn/01d@2x.png",
      }
    }
    style={
      {
        "height": 100,
        "width": 100,
      }
    }
  />
  <Text
    style={
      [
        {
          "fontSize": 32,
          "fontWeight": "500",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    21
    Â°C
  </Text>
  <Text
    style={
      [
        {
          "fontSize": 18,
          "fontWeight": "500",
          "textTransform": "capitalize",
        },
        {
          "color": "#000",
        },
      ]
    }
  >
    clear sky
  </Text>
</View>
`;
